hail/python/hail/ir/ir.py

```diff
@@ -1003,19 +1003,20 @@ def _compute_type(self, env, agg_env):
 
 
 class AggFilter(IR):
-    @typecheck_method(cond=IR, agg_ir=IR)
-    def __init__(self, cond, agg_ir):
+    @typecheck_method(cond=IR, agg_ir=IR, is_scan=bool)
+    def __init__(self, cond, agg_ir, is_scan):
```

Comment:
Do you want to delete ApplyAggOp and make give it an is_scan argument, too?  You know, for beautiful symmetry?

Reply:
Good idea. This might introduce some rebase headache, OK if I do this as part 3?

Reply:
Yes, of course.